### ISSUE TITLE
Button to get additional replies from remote instance

### ISSUE THREAD

**Issue by lucahammer** [elk-zone/elk#996]
### Clear and concise description of the problem

As a user I want to see all replies to a post so that I know if what I want to add has already been said or not. At the moment only replies my home instance knows about are displayed.

### Suggested solution

In the menu of the single status view we could provide an option to load more replies. Maybe in https://github.com/elk-zone/elk/blob/main/components/status/StatusActionsMore.vue

If the remote instance supports the Mastodon API, we can use the statuses/:id/context endpoint (https://docs.joinmastodon.org/methods/statuses/#context) to request up to 60 replies (without a Bearer token). Those can either be directly rendered and/or added to the database of the home instance through the search endpoint (https://docs.joinmastodon.org/methods/search/ with resolve parameter) so that other users see them without explicitly requesting them.

### Additional context

Mastodon should implement such an option through the API and do the logic on the server. But it currently doesn't and from the open issues it's unlikely to be done soon. 
https://github.com/mastodon/mastodon/issues/18150
https://github.com/mastodon/mastodon/issues/22704
https://github.com/mastodon/mastodon/issues/22674
https://github.com/mastodon/mastodon/issues/21497
https://github.com/mastodon/mastodon/issues/9409
https://github.com/mastodon/mastodon/issues/34

Doing it from the client is only a work around, but it would improve the user experience and if Mastodon implements it some day, the UI could stay the same and only the logic get replaced.

**Comment by KaKi87** [comment]
Hello,

Until this feature is officially added, I created [a userscript](https://git.kaki87.net/KaKi87/userscripts/src/branch/master/enhancementsForElkZone/README.md) that implements it :

![](https://git.kaki87.net/KaKi87/userscripts/media/branch/master/enhancementsForElkZone/screenshot1.gif)

Enjoy

**Comment by lucahammer** [comment]
@KaKi87 Thanks! I had to change some things to make it work for me (Firefox 115.0.2). Commenting here because I never got my account confirmation mail on your Forgejo instance. Leaving it here as it may help when implementing the feature and for other users of the userscript.

- Line 2 only matches the canary version of elk. I removed "main" to make it work on elk.zone
- Line 11 gave me a content security policy warning, so I copied the content of the createFetchInterceptor script to the userscript itself and removed the require.
- Line 41 prevents it from being triggered when opening Posts without reloading the page. I added "isDone=false;" after line 37 to change trigger it every time a replies are retrieved from the server.
- Line 54 to 74 gave me an error message for some reason, so I moved the loading of the script and css for the iziToast notifications to the top of the script. Without the promise, just "GM.addElement..."
- Line 104 prevented the script from loading other replies when one of them failed. In my case it failed with a reply from a GoToSocial instance. Haven't investigated the source of the error, but removed Line 104 with the return, to allow other replies to be fetched.

I am quite happy with it and hope it can be added directly to elk some day.

**Comment by shuuji3** [comment]
This issue still persists (ref. https://github.com/mastodon/mastodon/issues/34). Maybe we can try a similar approach as Phanpy's experimental feature (switching the status view to the original instance): https://github.com/cheeaun/phanpy/blob/b8f8271645136d4c45c4fdf5439d9da9c972ea3f/src/pages/status.jsx#L1260-L1282

This is not a perfect solution. Sometimes, the original server cannot allow unauthenticated access to public status, but that should improve usability for some cases by showing the status in the original instance within Elk, instead of opening the original server in a separate browser tab.

![phanpy's status screenshot showing "switch to post's instance" menu item](https://github.com/user-attachments/assets/7fc637b2-acc6-42dd-84c8-fbf94241ee3c)

![phanpy's status screenshot showing the panel to navigate to "switch to my instance"](https://github.com/user-attachments/assets/81da7f48-d0ee-462c-b57c-f743664e193a)

**Comment by KaKi87** [comment]
Thank you for making me discover this client though, it's awesome !

Especially with the way boosts are compacted and reply trees are displayed.

**Comment by shuuji3** [comment]
Yes, Phanpy has many interesting features. "Catch-up" is the unique one (https://phanpy.social/#/catchup) which collect recent posts and reorganize to make a temporary feed.

Both Elk and Phanpy uses the same Mastodon library masto.js underneath so technically Elk can do similar things.

**Comment by KaKi87** [comment]
> "Catch-up" is the unique one

Yeah, although this still suffers from Mastodon's inability to follow instances but only profiles and hashtags, which is why I made [MastoVue++](https://git.kaki87.net/KaKi87/mastovuepp/src/branch/master/README.md) for that purpose.
